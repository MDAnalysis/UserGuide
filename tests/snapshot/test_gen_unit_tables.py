
from doc.source.scripts.gen_unit_tables import write_unit_table
import tempfile

def test_write_unit_table(snapshot):
    with tempfile.NamedTemporaryFile() as f:
        tables = write_unit_table(f.name)
        assert tables == snapshot
