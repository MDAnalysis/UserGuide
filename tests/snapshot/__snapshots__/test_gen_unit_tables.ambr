# serializer version: 1
# name: test_write_unit_table
  list([
    tuple(
      'length',
      list([
        tuple(
          'A',
          1.0,
        ),
        tuple(
          'Angstrom',
          1.0,
        ),
        tuple(
          'angstrom',
          1.0,
        ),
        tuple(
          'femtometer',
          100000.0,
        ),
        tuple(
          'fm',
          100000.0,
        ),
        tuple(
          'nanometer',
          0.1,
        ),
        tuple(
          'nm',
          0.1,
        ),
        tuple(
          'picometer',
          100.0,
        ),
        tuple(
          'pm',
          100.0,
        ),
        tuple(
          'Å',
          1.0,
        ),
      ]),
    ),
    tuple(
      'density',
      list([
        tuple(
          'A^{-3}',
          1.0,
        ),
        tuple(
          'Angstrom^{-3}',
          1.0,
        ),
        tuple(
          'Molar',
          1660.5389210321898,
        ),
        tuple(
          'SPC',
          30.37184690488927,
        ),
        tuple(
          'TIP3P',
          29.85655608913766,
        ),
        tuple(
          'TIP4P',
          29.886382818497438,
        ),
        tuple(
          'nanometer^{-3}',
          1000.0,
        ),
        tuple(
          'nm^{-3}',
          1000.0,
        ),
        tuple(
          'water',
          30.00628806551247,
        ),
        tuple(
          'Å^{-3}',
          1.0,
        ),
      ]),
    ),
    tuple(
      'time',
      list([
        tuple(
          'AKMA',
          20.45482949774598,
        ),
        tuple(
          'femtosecond',
          1000.0,
        ),
        tuple(
          'fs',
          1000.0,
        ),
        tuple(
          'microsecond',
          1e-06,
        ),
        tuple(
          'millisecond',
          1e-09,
        ),
        tuple(
          'ms',
          1e-09,
        ),
        tuple(
          'nanosecond',
          0.001,
        ),
        tuple(
          'ns',
          0.001,
        ),
        tuple(
          'picosecond',
          1.0,
        ),
        tuple(
          'ps',
          1.0,
        ),
        tuple(
          's',
          1e-12,
        ),
        tuple(
          'sec',
          1e-12,
        ),
        tuple(
          'second',
          1e-12,
        ),
        tuple(
          'us',
          1e-06,
        ),
        tuple(
          'μs',
          1e-06,
        ),
      ]),
    ),
    tuple(
      'charge',
      list([
        tuple(
          'Amber',
          18.2223,
        ),
        tuple(
          'As',
          1.602176565e-19,
        ),
        tuple(
          'C',
          1.602176565e-19,
        ),
        tuple(
          'e',
          1.0,
        ),
      ]),
    ),
    tuple(
      'speed',
      list([
        tuple(
          'A/AKMA',
          0.04888821,
        ),
        tuple(
          'A/fs',
          1000.0,
        ),
        tuple(
          'A/ms',
          1e-09,
        ),
        tuple(
          'A/ps',
          1.0,
        ),
        tuple(
          'A/us',
          1e-06,
        ),
        tuple(
          'Angstrom/AKMA',
          0.04888821,
        ),
        tuple(
          'Angstrom/femtosecond',
          1000.0,
        ),
        tuple(
          'Angstrom/fs',
          1000.0,
        ),
        tuple(
          'Angstrom/microsecond',
          1e-06,
        ),
        tuple(
          'Angstrom/millisecond',
          1e-09,
        ),
        tuple(
          'Angstrom/ms',
          1e-09,
        ),
        tuple(
          'Angstrom/picosecond',
          1.0,
        ),
        tuple(
          'Angstrom/ps',
          1.0,
        ),
        tuple(
          'Angstrom/us',
          1e-06,
        ),
        tuple(
          'Angstrom/μs',
          1e-06,
        ),
        tuple(
          'angstrom/femtosecond',
          1000.0,
        ),
        tuple(
          'angstrom/fs',
          1000.0,
        ),
        tuple(
          'angstrom/microsecond',
          1e-06,
        ),
        tuple(
          'angstrom/millisecond',
          1e-09,
        ),
        tuple(
          'angstrom/ms',
          1e-09,
        ),
        tuple(
          'angstrom/picosecond',
          1.0,
        ),
        tuple(
          'angstrom/us',
          1e-06,
        ),
        tuple(
          'angstrom/μs',
          1e-06,
        ),
        tuple(
          'm/s',
          100.0,
        ),
        tuple(
          'nanometer/picosecond',
          0.1,
        ),
        tuple(
          'nanometer/ps',
          0.1,
        ),
        tuple(
          'nm/ns',
          100.0,
        ),
        tuple(
          'nm/ps',
          0.1,
        ),
        tuple(
          'pm/ps',
          100.0,
        ),
        tuple(
          'Å/ps',
          1.0,
        ),
      ]),
    ),
    tuple(
      'force',
      list([
        tuple(
          'J/m',
          1.66053892103219e-11,
        ),
        tuple(
          'N',
          1.66053892103219e-11,
        ),
        tuple(
          'Newton',
          1.66053892103219e-11,
        ),
        tuple(
          'kJ/(mol*A)',
          1.0,
        ),
        tuple(
          'kJ/(mol*Angstrom)',
          1.0,
        ),
        tuple(
          'kJ/(mol*nm)',
          10.0,
        ),
        tuple(
          'kJ/(mol*Å)',
          1.0,
        ),
        tuple(
          'kcal/(mol*Angstrom)',
          0.2390057361376673,
        ),
      ]),
    ),
    tuple(
      'energy',
      list([
        tuple(
          'J',
          1.66053892103219e-21,
        ),
        tuple(
          'eV',
          0.01036426919046959,
        ),
        tuple(
          'kJ/mol',
          1.0,
        ),
        tuple(
          'kcal/mol',
          0.2390057361376673,
        ),
      ]),
    ),
  ])
# ---
